server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/dto/QuotaDTO.java

```diff
@@ -102,5 +107,13 @@ public String toString() {
             .toString();
     }
 
+    @JsonIgnore
+    public boolean isValid() {
```

Comment:
I think instead, you can use Precondition checks in the Constructor to force the clients always create valid DTOs

Reply:
I thought of that but i don't want to do that. Because it would cause problems if there is already invalid quotas stored in the db.

Reply:
So, it's not done at the domain object level, in order to avoid breaking things for the users who already have negative quotas (if such user exists)
Not done in the DTO, because we don't want to break the api for getting the current quotas  for the users who already have negative quotas (if such user exists)

So the validation of the input from the API seems like a sane choice IHMO

Reply:
Why not created a `ValidQuotaDTO`?

I do not see what good computation can comes with an invalid `Quota`.

Reply:
After a good discussion with @mbaechler,
We proposed to : 
- do the validation at the domain object level
- add a sanitation phase when reading the value at the DAO level where we would replace the negative values (<-1) with 0. This way we will do the migration from the invalid values which did not allow to write, to a valid value which keep the same behavior.

WDYT ?